core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -542,7 +551,8 @@ class Partition(val topicPartition: TopicPartition,
         assignment = partitionState.replicas.asScala.map(_.toInt),
         isr = isr,
         addingReplicas = addingReplicas,
-        removingReplicas = removingReplicas
+        removingReplicas = removingReplicas,
+        LeaderRecoveryState.RECOVERED
```

Comment:
Yes and no. Unclean leader election is not possible when there is only one replica since the ISR set always equals the replica set. That means that if the replica is online then it will be elected as part of regular election and if it is offline the unclean leader election is a noop.

Reply:
I'll log a message.